### ISSUE TITLE
file injection synchronization with status completed during force-complete

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#7306]
Looks like the force-complete leaves files not injected far after the fact.
current situation is vlimant_BPH-Summer12-00228_00270_v0__161005_224414_1220 

These 187 files are missing in dbs
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/E2CACD0E-4995-E611-B821-001F29088E72.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/D8D7DA73-7495-E611-8203-842B2B760921.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/50F9F1BF-B395-E611-8B7B-44A8423C4026.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/90001/067C71E9-0E95-E611-B3CE-FA163E6B1E8B.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/A8A4EECC-9795-E611-8128-FA163E5C23E3.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/E28DC4F8-1895-E611-9001-3417EBE70684.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/3CF3C856-D994-E611-9F15-0CC47A7FC844.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/58CF5469-3395-E611-9793-00259021A306.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/1E2ECB23-5695-E611-B21F-549F35AE4FBC.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/90001/42CCD703-DF94-E611-8E25-00215E2EB778.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/EAC25F06-CE95-E611-9EA6-002590CB0B5A.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/18EEF259-5795-E611-B3DE-1CC1DE18CD26.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/50BF5365-6495-E611-AD21-0025905C3E68.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/100000/049307BE-8E95-E611-A067-0025901D40A6.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/1C1BB87D-7495-E611-A281-001E673C808A.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/2E946EFC-AE95-E611-8180-0CC47A7C34E6.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/922F32FA-0C95-E611-A5D7-00215A491956.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/442E9CAF-7D95-E611-949E-0CC47AA9943A.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/6A31A796-3A95-E611-B30D-008CFA1C9308.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/04EC5240-B395-E611-8D77-7845C4FC36E0.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/70000/461CC4F1-C395-E611-9ABC-002590909086.root
/store/mc/Summer12/X5568ToBsPi_Mod2BMuonFilter_TuneZ2star_8TeV-pythia6-evtgen/GEN-SIM/START53_V7C-v2/80001/12A01943-5F95-E611-B355-44A842B2D631.root 
...

operationally, files need to have been injected when the workflow comes to completed.

**Comment by vlimant** [comment]
ping on this. We have workflows that are not getting through with their files since many days.
@sidnarayanan could this be also files removed from TMDB but not from DBS because not yet in DBS ?
FYI @scarletnorberg @mcremone

**Comment by sidnarayanan** [comment]
I checked a few of these files and they are present in TMDB. Investigating a little bit, it looks like there are two unrelated issues: timeouts at JINR and data loss at T2_UK_SGrid_RALPP. Opening a ticket for the first issue and checking the file is actually missing for the second.

**Comment by vlimant** [comment]
adding pdmvserv_BPH-RunIISummer15GS-00085_00436_v0__161003_004347_9946

**Comment by vlimant** [comment]
adding pdmvserv_TOP-RunIISummer15wmLHEGS-00100_00187_v0__161006_001616_6712 which was also force-complete

**Comment by vlimant** [comment]
@ticoann @amaltaro do you want another GH issue to focus on the force-complete->completed state transition ?
this is rather annoying and slowing things down

**Comment by ticoann** [comment]
Sorry Jean-Roch, I will look into the problem.

**Comment by vlimant** [comment]
that'd be great thx
@prozober

**Comment by vlimant** [comment]
Can you guys please do something about 
pdmvserv_task_BTV-RunIISummer15GS-00069__v1_T_161125_213521_3712
pdmvserv_task_BTV-RunIISummer15GS-00070__v1_T_161125_213537_5698
pdmvserv_task_BTV-RunIISummer15GS-00074__v1_T_161125_213652_5819
pdmvserv_task_SUS-RunIISummer15GS-00196__v1_T_161125_233700_7116

**Comment by amaltaro** [comment]
Jean-Roch, where are files missing? In Phedex or DBS? For which dataset(s)? Can you also point me to that unified page with the history of the close-out checks (which contains files for each output dataset and so on for a workflow). It should help a lot debugging this problem.

**Comment by vlimant** [comment]
@amaltaro  http://dabercro.web.cern.ch/dabercro/unified/showlog/?search=BTV-RunIISummer15GS-00069 and the like. For this one, agent 311 is still running a merge.

i.e. not all agents do get to complete the wf.

**Comment by amaltaro** [comment]
Thanks! besides (running only logcollect, which btw seems to be consistent now)
 pdmvserv_task_SUS-RunIISummer15GS-00196__v1_T_161125_233700_7116

ALL the other workflows are still running a few processing or merge jobs, and some are actually in completed status.(?) So the real problem seems to be force-completing workflows in the agents instead of syncing files/blocks betwen dbs/phedex.

**Comment by vlimant** [comment]
pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031 was just toggled force-complete

**Comment by amaltaro** [comment]
Ok, I thought it could be a race condition between failing the file subscriptions (available/acquired) and the JobCreator creating jobs. 

However, this workflow you just force-completed was moved to automatically moved to completed in a short time (maybe < 10min). This transition was never seen by vocms0309, the last entry in WorkQueueManager logs dates back to Nov 26.

There are 3 jobs still in the condor queue (and for several hours)
```
cmst1@vocms0309:/data/srv/wmagent/current $ condorq -const 'WMAgent_RequestName == "pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031"'
53266.3  2  21449 /data/srv/wmagent/v1.0.21.patch1/install/wmagent/JobCreator/JobCache/pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031/BTV-RunIISummer15GS-00073_0/BTV-RunIISummer15GS-00073_0MergeRAWSIMoutput/BTV-RunIISummer16DR80Premix-00075_0/JobCollection_304884_0/job_3155128/condor.53266.3.log T2_DE_RWTH
53279.4  2  20318 /data/srv/wmagent/v1.0.21.patch1/install/wmagent/JobCreator/JobCache/pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031/BTV-RunIISummer15GS-00073_0/BTV-RunIISummer15GS-00073_0MergeRAWSIMoutput/BTV-RunIISummer16DR80Premix-00075_0/JobCollection_304884_0/job_3155127/condor.53279.4.log T2_DE_RWTH
53312.0  2  17303 /data/srv/wmagent/v1.0.21.patch1/install/wmagent/JobCreator/JobCache/pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031/BTV-RunIISummer15GS-00073_0/BTV-RunIISummer15GS-00073_0MergeRAWSIMoutput/BTV-RunIISummer16DR80Premix-00075_0/JobCollection_304884_0/job_3155126/condor.53312.0.log T2_DE_RWTH
cmst1@vocms0309:/data/srv/wmagent/current $ 
```

and finally, the WQE (and GQE) are gone:
```
cmst1@vocms0309:/data/srv/wmagent/current $ python getWQStatusByWorkflow.py pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031

Summary for https://cmsweb.cern.ch/couchdb/workqueue and request pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031
{'numberOfElements': 0}

Summary for http://localhost:5984/workqueue and request pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031
{'numberOfElements': 0}

Summary for http://localhost:5984/workqueue_inbox and request pdmvserv_task_BTV-RunIISummer15GS-00073__v1_T_161125_213646_6031
{'numberOfElements': 0}
```

meaning this workflow will run until the end of its life in vocms0309.

Bottom line (probably applies for aborting wfs as well), the problem lies on the synchronization between central services and all the agents serving this request. 

Seangchan and I discussed it in the past week and there is a plan to fix it for good.

**Comment by amaltaro** [comment]
Ok, these workflows were force-killed inside each agent. There were a very few jobs still running on vocms0309 and vocms0311 (non cleanup or logcllect).

**Comment by amaltaro** [comment]
... and according to a simple script I just wrote, all their outputs are consistent between DBS and PhEDEx (in terms of number of files per dataset), for all those 5 workflows you mentioned above.
This means the agent properly injects/closes blocks and files into those DBs.

What's left is getting the proper transition into all the agents. Seangchan, let me know if you need my help on this, though I can only tackle it in the next week :)

**Comment by vlimant** [comment]
thx, glad that in the meantime things were able to close by themselves.
Looking forward to the fix.

**Comment by vlimant** [comment]
can pdmvserv_task_BTV-RunIISummer15GS-00072__v1_T_161125_213611_6014 please be put in the proper status in the agent ASAP

**Comment by amaltaro** [comment]
pdmvserv_task_BTV-RunIISummer15GS-00072__v1_T_161125_213611_6014
was force-killed in all the agents. vocms0126 and vocms0130 still had a couple of jobs in condor.

**Comment by vlimant** [comment]
thx !

**Comment by amaltaro** [comment]
Summary of this discussion is that force-complete workflows were still injecting data into wmbs and creating new jobs, that's why DBS and TMDB was not in sync. This #7442 should have fixed this issue.

In any case, also somehow related to #7115. So closing this one
